front/lib/resources/vault_resource.ts

```diff
@@ -172,22 +173,38 @@ export class VaultResource extends BaseResource<VaultModel> {
     return new this(VaultModel, vault.get());
   }
 
+  static async fetchByName(
```

Comment:
Not a big fan of exposing this method if we can hide this when we creating the vault 🙏.